### ISSUE TITLE
Rule proposal: disallow `process.argv.splice(2)[0]`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#3074]
### Description

I found

```js
const foo = process.argv.splice(2)[0];
```

in a repo.

### Examples

```js
// ❌
const foo = process.argv.splice(2)[0];

// ✅
const foo = process.argv.slice(2)[0];

// ✅
const foo = process.argv[2];
```

### Proposed rule name

not sure

### Additional Info

I'm not sure how this rule should work, 

1. splice + index access?
2. splice with only "start"  argument?
3. Only check `process.argv.splice`?

**Comment by sindresorhus** [comment]
Wouldn't this be better proposed in https://github.com/eslint-community/eslint-plugin-n ? If we add it, and they eventually add it, we will have to remove it, so it seems better to add it there in the first place.

**Comment by fisker** [comment]
I was wondering if the array is not `process.argv` but just a normal array.

Should we report?
